### PR TITLE
session change issues sorted. also try and use laps compleated for to…

### PR DESCRIPTION
**Pull request by Sparten**
…tal distance driven and some other multiclass char ahead/behind changes

### CODE REVIEW

File: CrewChiefV4/iRacing/Sim.cs [hunk 1]
```diff
@@ -37,52 +34,35 @@ public Sim()
...
-                Console.WriteLine("MustReloadDrivers: true");
+                Console.WriteLine("Reloading Drivers");
                 _drivers.Clear();
                 _driver = null;
-                _paceCar = null;
-                _mustReloadDrivers = false;
             }
 
-            // Assume max 70 drivers
+            // Assume max 128 drivers
```
Comment (mrbelowski) [inline CrewChiefV4/iRacing/Sim.cs]: this comment suggests the for loop should go up to 128, but it doesn't - am i missing something?

File: CrewChiefV4/iRacing/Sim.cs [hunk 2]
```diff
@@ -91,19 +71,12 @@ private void GetDrivers(SessionInfo info)
                 {
                     driver = Driver.FromSessionInfo(info, id);
 
-                    // If no driver found, end of list reached
                     if (driver == null)
                     {
                         continue;
                     }
 
                     driver.IsCurrentDriver = false;
```
Comment (mrbelowski) [inline CrewChiefV4/iRacing/Sim.cs]: now we're including the pace car in the array, are you confident it doesn't trigger weird messages?

Comment (Sparten) [reply CrewChiefV4/iRacing/Sim.cs]: Yes it should be ruled out in the opponent array

File: CrewChiefV4/iRacing/Sim.cs (hunk 1 above)
Comment (Sparten) [reply CrewChiefV4/iRacing/Sim.cs]: ups me testing

File: CrewChiefV4/iRacing/iRacingGameStateMapper.cs [hunk 3]
```diff
@@ -121,14 +121,8 @@ public GameStateData mapToGameStateData(Object memoryMappedFileStruct, GameState
...
-                (currentGameState.SessionData.SessionType == SessionType.Practice && lastSessionType == SessionType.Practice) ||
-                (currentGameState.SessionData.SessionType == SessionType.Qualify && lastSessionType == SessionType.Qualify)) &&
-                ((lastSessionPhase == SessionPhase.Green || lastSessionPhase == SessionPhase.FullCourseYellow) || lastSessionPhase == SessionPhase.Finished) &&
-                (currentGameState.SessionData.SessionPhase == SessionPhase.Countdown);
-
-            if (sessionOfSameTypeRestarted || currentGameState.SessionData.SessionType != SessionType.Unavailable
-                && lastSessionPhase != SessionPhase.Countdown
+
+            if (currentGameState.SessionData.SessionType != SessionType.Unavailable && shared.Telemetry.IsNewSession
                 && lastSessionType != currentGameState.SessionData.SessionType || sessionNumber != previousSessionNumber || previousSessionId != currentGameState.SessionData.SessionId)
```
Comment (mrbelowski) [inline CrewChiefV4/iRacing/iRacingGameStateMapper.cs]: might be worth putting brackets around these so it's
if (currentGameState.SessionData.SessionType != SessionType.Unavailable && shared.Telemetry.IsNewSession
                  && (lastSessionType != currentGameState.SessionData.SessionType || sessionNumber != previousSessionNumber || previousSessionId != currentGameState.SessionData.SessionId))

Unless I'm mis-reading the logic

**Pull request merged**